source/Calamari.Shared/Integration/Scripting/Python/PythonScriptEngine.cs

```diff
@@ -5,44 +5,34 @@
 
 namespace Calamari.Integration.Scripting.Python
 {
-    public class PythonScriptEngine : IScriptEngine
+    public class PythonScriptEngine : ScriptEngine
     {
-        public ScriptSyntax[] GetSupportedTypes()
+        public override ScriptSyntax[] GetSupportedTypes()
         {
             return new[] {ScriptSyntax.Python};
         }
 
-        public CommandResult Execute(
-            Script script,
-            CalamariVariableDictionary variables,
-            ICommandLineRunner commandLineRunner,
+        protected override IEnumerable<ScriptExecution> PrepareExecution(Script script, CalamariVariableDictionary variables,
```

Comment:
`IEnumerable<ScriptExecution>` bothers me a bit because I am expecting just a single execution. I understand it's required to run the python dependency script. I don't have a good suggestion for an alternative.

Reply:
I wonder if we (eventually) need to refactor the python dependency install and script execution into the one call somehow. Food for thought.

Reply:
Either that, or we introduce another method for "preparing an execution for installing dependencies"?
We could just not override it on the other methods? :thinking:
Stuff for the future.